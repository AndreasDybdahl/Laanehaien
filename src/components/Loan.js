import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { appIntro, form, btn } from '../styles/App.css';

class Loan extends Component {

    constructor() {
        super();
        this.state = {
            value: 5,
            maxLoan: 500000,
        };
    }

    getLoanValue() {
        const {
            value,
            maxLoan
            } = this.state;
        return value * maxLoan;
    }

    handleChange(e) {
        this.setState({
            value: e.target.value
        })
    }

    handleSubmit(e) {
        e.preventDefault();
        this.props.postLoan({amount: this.getLoanValue()});
    }

    renderLoan(loan) {
        return loan.toString().replace(/\B(?=(\d{3})+(?!\d))/g, " ");
    }

    render() {
        return (
            <div className={classnames(appIntro)}>
                <form className={classnames(form)} onSubmit={e => this.handleSubmit(e)}>
                    <br /><br />
                    <input style={{width: 277}} type="range" name="points" min="0.01" max="12" step="0.2" value={this.state.value} onChange={e => this.handleChange(e) }/>
                    <br />
                    <br />
                    Jeg vil låne { this.renderLoan(parseInt(this.state.value * this.state.maxLoan, 0)) },- kr
                    <br />
                    <br />
                    <button className={classnames(btn)} type="submit">Søk om lån</button>
                </form>
            </div>
        );
    }
}

Loan.propTypes = {
    postLoan: PropTypes.func
};

export default Loan;