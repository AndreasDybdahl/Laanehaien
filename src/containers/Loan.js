import {connect} from 'react-redux';
import Loan from '../components/Loan';
import {postLoan} from '../modules/loans';

export default connect(
    null,
    {
        postLoan
    }
)(Loan);