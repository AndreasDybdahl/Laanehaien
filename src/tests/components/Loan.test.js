import { shallow, mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Loan from '../../components/Loan';


describe('Loan', () => {

it('component should render', () => {
    const tree = renderer.create(
        <Loan />
    ).toJSON();

    expect(tree).toMatchSnapshot();
})

it('component should render with correct values', () => {
     const component = renderer.create(
        <Loan />
    );

    let tree = component.toJSON();

    const form = tree.children.filter(child => child.type === 'form')[0];

    const input = form.children.filter(child => child.type === 'input')[0];

    input.props.onChange({
        target: {
            value: '7.61' // 3 805 000
        }
    })

    tree = component.toJSON();

    const changes = tree.children.filter(child => child.type === 'form')[0];

    const valueText = changes.children.filter(child => child.type === 'p')[0];

    const value = valueText.children.filter(child => child.type === 'span')[0]

    expect(value.children[0]).toEqual('3 805 000');
})

it('default state with enzyme', () => {
   const component = shallow(<Loan />);
   expect(component.find('[name="amount"]').text()).toEqual('2 500 000,- kr')
})


it('component should render with correct values, risks is checked and form is submitted (enzyme)', () => {
   const component = mount(<Loan />);
   const input = component.find('input').first();
   const checkbox = component.find('[name="doRisk"]').first();
   const button = component.find('button').first();

   input.simulate('change', {target: {value: '8.01'}});
   checkbox.simulate('change', {target: {checked: true}});
   expect(component.find('[name="amount"]').text()).toEqual('4 005 000,- kr');
   expect(button.hasClass('btnDisabled')).toEqual(false);

})

});