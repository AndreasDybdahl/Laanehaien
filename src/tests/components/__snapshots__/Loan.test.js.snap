exports[`Loan component should render 1`] = `
<div
  className="">
  <form
    className=""
    onSubmit={[Function]}>
    <br />
    <br />
    <input
      max="12"
      min="0.01"
      name="points"
      onChange={[Function]}
      step="0.2"
      style={
        Object {
          "width": 277,
        }
      }
      type="range"
      value={5} />
    <p>
      Jeg vil låne 
      <span
        name="amount">
        2 500 000
        ,- kr
      </span>
    </p>
    <div
      className="">
      <h3
        style={
          Object {
            "margin": 0,
          }
        }>
        Risiko:
      </h3>
      <p
        style={
          Object {
            "marginTop": 0,
          }
        }>
        (Om man ikke betaler)
      </p>
      <ul
        style={
          Object {
            "margin": "0 auto",
            "width": 150,
          }
        }>
        <li>
           
          Brukket tann
        </li>
        <li>
           
          Knekt ben
        </li>
      </ul>
    </div>
    <p>
      <label>
        <span>
          Jeg godtar risikoen
        </span>
        <input
          name="doRisk"
          onChange={[Function]}
          type="checkbox" />
      </label>
    </p>
    <button
      className="btn btnDisabled"
      disabled={true}
      type="submit">
       Søk om lån
    </button>
  </form>
</div>
`;
